src/code/data/migrations/24_fix_simulation_type.coffee

```diff
@@ -0,0 +1,18 @@
+AppSettingsStore = require('../../stores/app-settings-store').store
+
+migration =
+  version: "1.23.0"
+  description: "Fix simulation type - 1.21.0 to 1.22.0 sim-type broken"
+  date: "2018-10-15"
+
+  doUpdate: (data) ->
+    data.settings ?= {}
+
+    # This migration addresses a failing case where a simulation with
+    # collector nodes was not coreectly migrating the simulationType
+    # to time-based.
+    
+    if data.nodes.some((node) -> node.data.isAccumulator)
```

Comment:
We use lodash throughout the project (its a global, set in globals.coffee).  I'm not sure that is needed anymore but for consistency I'd replace this with:

`if _.some(data.nodes, (node) -> node.data.isAccumulator)`